xcfun: init at 2.1.1

sheepforce opened the pull request:
###### Motivation for this change
Adds xcfun, which is a missing dependency for PySCF (quantum chemistry in python). Part of the packages @markuskowa and me want to merge from markuskowa/NixOS-QChem#56 .

(I am not very confident where to put libraries like this. For now I've put it into `development/libraries/science/chemistry`)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/development/libraries/science/chemistry/xcfun/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  cmakeFlags = [ "-DXCFUN_MAX_ORDER=3" ];
+
+  enableParallelBuilding = true;
+
```
```suggestion
```
set by cmake.

SuperSandro2000 commented on pkgs/development/libraries/science/chemistry/xcfun/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub  {
+    owner = "dftlibs";
+    repo = pname;
+    rev = "v${version}";
+    sha256= "1bj70cnhbh6ziy02x988wwl7cbwaq17ld7qwhswqkgnnx8rpgxid";
+  };
+
```
```suggestion
```

SuperSandro2000 commented on pkgs/development/libraries/science/chemistry/xcfun/default.nix:
```diff
@@ -0,0 +1,35 @@
+in stdenv.mkDerivation rec {
+  pname = "xcfun";
+  version = "2.1.1";
+
```
```suggestion

  src = fetchFromGitHub  {
    owner = "dftlibs";
    repo = pname;
    rev = "v${version}";
    sha256= "1bj70cnhbh6ziy02x988wwl7cbwaq17ld7qwhswqkgnnx8rpgxid";
  };

```

SuperSandro2000 commented on pkgs/development/libraries/science/chemistry/xcfun/default.nix:
```diff
@@ -0,0 +1,35 @@
+{ stdenv, lib, fetchFromGitHub, cmake, gfortran, python3 } :
+
+let
+  python = python3.withPackages (p: with p; [ pybind11 ]);
+
+in stdenv.mkDerivation rec {
```
```suggestion
stdenv.mkDerivation rec {
```

SuperSandro2000 commented on pkgs/development/libraries/science/chemistry/xcfun/default.nix:
```diff
@@ -0,0 +1,35 @@
+    gfortran
+  ];
+
+  propagatedBuildInputs = [ python ];
```
```suggestion
  propagatedBuildInputs = [ (python3.withPackages (p: with p; [ pybind11 ])) ];
```

r-rmcgibbo commented:
Result of `nixpkgs-review pr 127378` at c0033749 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>xcfun</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unnecessary-parallel-building.md">unnecessary-parallel-building</a>

Meson, CMake and qmake already set `enableParallelBuilding = true` by default so it is not necessary.

Near pkgs/development/libraries/science/chemistry/xcfun/default.nix:19:3:
```
   |
19 |   enableParallelBuilding = true;
   |   ^
```
</li>
  </ul>
</details>

sheepforce committed:
xcfun: init at 2.1.1


Update pkgs/development/libraries/science/chemistry/xcfun/default.nix

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
Update pkgs/development/libraries/science/chemistry/xcfun/default.nix

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
Update pkgs/development/libraries/science/chemistry/xcfun/default.nix

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
Update pkgs/development/libraries/science/chemistry/xcfun/default.nix

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
Update pkgs/development/libraries/science/chemistry/xcfun/default.nix

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>
